api/providers/views.py

```diff
@@ -205,26 +186,97 @@ class PreprintProviderHighlightedSubjectList(JSONAPIBaseView, generics.ListAPIVi
     serializer_class = TaxonomySerializer
 
     def get_queryset(self):
-        provider = get_object_or_error(PreprintProvider, self.kwargs['provider_id'], self.request, display_name='PreprintProvider')
+        provider = get_object_or_error(self._model_class, self.kwargs['provider_id'], self.request)
         return optimize_subject_query(Subject.objects.filter(id__in=[s.id for s in provider.highlighted_subjects]).order_by('text'))
 
 
-class PreprintProviderLicenseList(LicenseList):
+class CollectionProviderHighlightedSubjectList(GenericProviderHighlightedSubjectList):
+    view_category = 'colleciton_providers'
+    _model_class = CollectionProvider
+
+
+class PreprintProviderHighlightedSubjectList(GenericProviderHighlightedSubjectList):
+    view_category = 'preprint_providers'
+    _model_class = PreprintProvider
+
+
+class GenericProviderLicenseList(LicenseList):
     """The documentation for this endpoint can be found [here](https://developer.osf.io/#operation/preprint_provider_licenses_list)
     """
     ordering = ()  # TODO: should be ordered once the frontend for selecting default licenses no longer relies on order
-    view_category = 'preprint_providers'
 
     def get_queryset(self):
-        provider = get_object_or_error(PreprintProvider, self.kwargs['provider_id'], self.request, display_name='PreprintProvider')
+        provider = get_object_or_error(self._model_class, self.kwargs['provider_id'], self.request)
```

Comment:
minor since `display_name` is just a nicety for error formating, but wouldn't `__name__` work here?

Reply:
It seems like that should be default in `get_object_or_error`, but I'll put it here